Add noscores/novalues option to zscan/hscan. (#2612)
redis/redis-doc: commands/scan.md

It is important to note that the **TYPE** filter is also applied after elements are retrieved from the database, so the option does not reduce the amount of work the server has to do to complete a full iteration, and for rare types you may receive no elements in many iterations.

## The NOSCORES/NOVALUES option

> When using `HSCAN`, you can use `NOVALUES` option to make Redis return only the keys in the hash table, without their corresponding values.

Reviewer: I hope you don't mind a little bit of copy-editing.

```suggestion
When using `HSCAN`, you can use the `NOVALUES` option to make Redis return only the keys in the hash table without their corresponding values.
```

> Similarly, when using `ZSCAN`, you can use `NOSCORES` option to make Redis return only the members in the zset, without their corresponding scores.

Reviewer: ```suggestion
Similarly, when using `ZSCAN`, you can use the `NOSCORES` option to make Redis return only the members in the sorted set without their corresponding scores.
```

```

> redis 127.0.0.1:6379> HMSET myhash a 1 b 2

Reviewer: Since HMSET is deprecated, let's use HSET.

```suggestion
redis 127.0.0.1:6379> HSET myhash a 1 b 2
```
